File: Zadania Szkopul (C++ i Java)/Sloneczny Robot/Sloneczny_Robot.cpp
#include <bits/stdc++.h>

using namespace std;
int n,b,a,maksa; //n = dlugosc odcinka, b - poj baterii, a - poj akumulatora, maksa - maksymalna pojemnosc akumulatora
int pt=0; //pt - pokonana trasa
int Swi[200009]; // tablica swiatla
int main()
{
	scanf("%d %d %d", &n, &b, &a);
	maksa=a;
	for(int i=1;i<=n;i++)
	{
		scanf("%d", &Swi[i]);
	}
	for(int i=1;i<=n;i++)
	{
		if(Swi[i]==1 and b!=0 and a<maksa)
		{
			b--;
			a++;
			pt++;
		}
		else if(a!=0)
		{
			a--;
			pt++;
		}
		else if(Swi[i]==0 and a==0 and b!=0)
		{
			b--;
			pt++;
		}
		else if(a==0 and b==0)
		{
			break;
		}
	}
	printf("%d", pt);
}

File: Zadania Szkopul (C++ i Java)/Sloneczny Robot/Sloneczny_Robot.java
import java.util.Scanner;

public class Sloneczny_Robot {
    public static void main(String[] args) {
        int n,b,a,maksa; //n = dlugosc odcinka, b - poj baterii, a - poj akumulatora, maksa - maksymalna pojemnosc akumulatora
        int pt=0; //pt - pokonana trasa
        int Swi[] = new int[200009]; // tablica swiatla
        Scanner czytaj = new Scanner(System.in);
        n = czytaj.nextInt();
        b = czytaj.nextInt();
        a = czytaj.nextInt();
        maksa = a;
        for(int i=1; i<=n; i++){
            Swi[i] = czytaj.nextInt();
        }
        czytaj.close();
        for(int i=1; i<=n; i++){
            if(Swi[i]==1 && b!=0 && a<maksa){
                b--;
                a++;
                pt++;
            }
            else if(a!=0){
                a--;
                pt++;
            }
            else if(Swi[i]==0 && a==0 && b!=0){
                b--;
                pt++;
            }
            else if(a==0 && b==0){
                break;
            }
        }
        System.out.println(pt);
    }
}
